server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -625,6 +625,7 @@ protected Node(
             final RerouteService rerouteService = new BatchedRerouteService(clusterService, clusterModule.getAllocationService()::reroute);
             rerouteServiceReference.set(rerouteService);
             clusterService.setRerouteService(rerouteService);
+            clusterService.setAllocationService(clusterModule.getAllocationService());
```

Comment:
`AllocationService` can be injected in `TransportGetAutoscalingCapacityAction`, so there's no need to modify `ClusterService`

Reply:
Hm, `ReactiveStorageDeciderService` gets created in `Autoscaling`, not in `TransportGetAutoscalingCapacityAction`, where we don't have access to the DI framework yet, so we have to pass `AllocationService` via `ClusterServce`.
Maybe I'm missing out something?

Reply:
That's correct, but you can still provide a reference in `AutoscalingDeciderService#scale`? I think that the `ClusterService` shouldn't provide the `AllocationService`, I feel it's confusing.